libCom/posix: add env var to opt out of FIFO scheduling

ralphlange opened the pull request:
- add EPICS_ALLOW_POSIX_THREAD_PRIORITY_SCHEDULING (default: YES)
- start EPICS threads with default policy when set to NO

fixes #734

ralphlange committed:
libCom/posix: don't memlock() if opted out of FIFO scheduling

ralphlange committed:
PR #737: add Release Note snippet

anjohnson commented on documentation/new-notes/PR-737.md:
```diff
@@ -0,0 +1,5 @@
+
+On POSIX systems, processes with real-time capabilities can opt out of using
+Posix thread priority scheduling and memory locking.
+Set EPICS_USE_POSIX_THREAD_PRIORITY_SCHEDULING=NO to achieve this.
```
Strict Markdown rules require literal underscore characters to be escaped, and in this case we want RTD to format that setting as code so please use back-ticks, like this:
```markdown
Set `EPICS_THREAD_ALLOW_PRIORITY_SCHEDULING=NO` in the environment to achieve this.
```

anjohnson commented on configure/CONFIG_ENV:
```diff
@@ -53,3 +53,5 @@ EPICS_IOC_IGNORE_SERVERS=""
 EPICS_IOC_LOG_PORT=7004
 
+# Posix priority scheduling
+EPICS_USE_POSIX_THREAD_PRIORITY_SCHEDULING=YES
```
With my naming Czar hat on, the name of this setting doesn't quite match what it does.

The build variable `USE_POSIX_THREAD_PRIORITY_SCHEDULING` that is set in `configure/CONFIG_SITE` controls whether the epicsThreads implementation should be built to allow suitably privileged IOCs to use priority scheduling. I question whether anyone actually disables that now (I suspect it was needed for some early Linux libc versions, we should check the core-talk archives), but that's a different topic and might become a Codeathon project to rip it out completely.

This new environment variable provides the ability to deny the use of priorities for code built on epicsThreads, but if the build was configured to disable them this can't force them to be used (so the name shouldn't say "use").

The comment should be something like "Allow/prevent POSIX priority scheduling of EPICS threads", and the variable name should be more like `EPICS_THREAD_ALLOW_POSIX_PRIORITY_SCHEDULING`

anjohnson reviewed (changes requested):
Name and doc changes

ralphlange committed:
libCom/posix: add env var to opt out of FIFO scheduling

- add EPICS_ALLOW_POSIX_THREAD_PRIORITY_SCHEDULING
  (default: YES)
- start EPICS threads with default policy
  when set to NO

anjohnson reviewed (approved):
Thanks, looks good now.

ralphlange commented:
Confirmed as working in the original setup by @anzez.

anjohnson commented:
The long environment variable name used here coupled with a recent change to iocStats unfortunately triggered [this issue](https://github.com/epics-modules/iocStats/issues/75).
